### PR TITLE
CRW-586 add Quarkus devfile +...

### PR DESCRIPTION
**Pull request by nickboldt**
CRW-586 add Quarkus devfile + icon

Change-Id: I9f2fb6cde23ecf3eefa5d04e5b7b58cf04a5228a
Signed-off-by: nickboldt <nboldt@redhat.com>

### CODE REVIEW

**Comment by nickboldt** [comment]
Failed to load devfile in OCP 4.1 instance (as in #222) but in che.openshift.io it loaded OK.

We should probably tweak the sample project so it's README.md doesn't mention JDK8 (as the CRW version is for JDK11.0.6 embedded in GraalVM).

**Comment by svor** [comment]
@nickboldt 
I tried native compilation in this devfile and got an error
```log
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.1.1.Final:native-image (default) on project getting-started: Failed to generate a native image: Failed to augment application classes: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.IllegalStateException: Unsupported version of GraalVM detected: GraalVM Version 19.3.1 CE. Please use GraalVM 19.2.1.
```
In https://github.com/quarkusio/quarkus/wiki/Migration-Guide-1.3 As I see GraalVM JDK 11 is a tech preview for now. So probably it isn't stable right now.
Also I found related issue https://github.com/quarkusio/quarkus/issues/5682
And Quarkus guide says that we need GraalVM version 19.2.1 - https://quarkus.io/guides/building-native-image#prerequisites.

So probably for now we have to use Java-8 and GraalVM-19.2.1 to provide native compilation

**Comment by nickboldt** [comment]
Hmm, odd. https://github.com/quarkusio/quarkus/wiki/Migration-Guide-1.2 doesn't mention that JDK 11 is tech preview (and that's the version we have in the plugin registry).

**Comment by tsmaeder** [comment]
@nickboldt since @svor did the upstream plugin, he's a much better reviewer than me. I can't request him, though.

**Comment by nickboldt** [comment]
But you could have assigned it to him?

**Comment by nickboldt** [comment]
graalvm & c/c++ stuff removed. openjdk 11 added.

[quay.io/crw/plugin-java11-rhel8:2.1-7(https://quay.io/crw/plugin-java11-rhel8:2.1-7)

File: dependencies/che-devfile-registry/devfiles/03_java11-quarkus/devfile.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  -
+    name: quarkus-quickstarts
+    source:
+      type: git
+      location: "https://github.com/quarkusio/quarkus-quickstarts.git"
+      sparseCheckoutDir: /getting-started/
+components:
+  -
+    type: chePlugin
+    id: redhat/quarkus-java8/latest
```
Comment (svor) [inline dependencies/che-devfile-registry/devfiles/03_java11-quarkus/devfile.yaml]: It should be redhat/quarkus-java11/latest 
see https://github.com/redhat-developer/codeready-workspaces/blob/master/dependencies/che-plugin-registry/v3/plugins/redhat/quarkus-java11/1.2.0/meta.yaml#L3

Comment (svor) [reply dependencies/che-devfile-registry/devfiles/03_java11-quarkus/devfile.yaml]: And `redhat/quarkus-java11/latest` should contain 
```yaml
      args:
        - sh
        - -c
        - ${PLUGIN_REMOTE_ENDPOINT_EXECUTABLE}
```
It isn't enough to have  `- ${PLUGIN_REMOTE_ENDPOINT_EXECUTABLE}` the error is 
```log
 Error: Failed to run the workspace: "The following containers have terminated:
vscode-quarkuse4w: reason = 'Error', exit code = 127, message = 'null'"
```
Probably it is related to all plugins., did you tested this changes: https://github.com/redhat-developer/codeready-workspaces/commit/82e1ab076ab85c817fa6478d2121242beae98a56

Comment (nickboldt) [reply dependencies/che-devfile-registry/devfiles/03_java11-quarkus/devfile.yaml]: crap. no. I just took @amisevsk 's advice :(

will revert that change in master and fix this PR

Comment (amisevsk) [reply dependencies/che-devfile-registry/devfiles/03_java11-quarkus/devfile.yaml]: My bad, I glanced at the dockerfiles for the sidecars and didn't realize that `CMD /entrypoint.sh` was the "shell form", which automatically prefixes `sh -c`.

Comment (nickboldt) [reply dependencies/che-devfile-registry/devfiles/03_java11-quarkus/devfile.yaml]: all good: both my first mistake, then my corrected (but still wrong) commit have been reverted to the old state from CRW 2.0.

**Pull request merged**